src/p256.rs

```diff
@@ -71,8 +70,7 @@ fn point_add(p: Jacobian, q: Jacobian) -> Jacobian {
 
     if u1 == u2 {
         if s1 == s2 {
-            assert!(false);
-            return point_double(p);
+            unreachable!();
```

Comment:
Hm, I'm not sure if we want to allow `unreachable`. Let's do it for now but we might end up not allowing it.
I opened issue #3 to keep track of this.

Reply:
`unreachable!()` means the same as `assert!(false)` – it's just slightly more clear.

(it's different from `unreachable_unchecked!()`)

Reply:
Right, we just like to keep the language as concise as possible so we have to decide if having this is worth it :slightly_smiling_face:

Reply:
Right, I was thinking from a Rust POV. The other way to approach this would be getting rid of the `if/else` in the first place, just putting `assert_ne!(s1, s2);` there.

Reply:
Getting rid of the `if/else` might actually be the nicest option....